spring-cloud-gcp-secretmanager/src/main/java/org/springframework/cloud/gcp/secretmanager/SecretManagerTemplate.java

```diff
@@ -0,0 +1,196 @@
+public class SecretManagerTemplate {
+
+	private final SecretManagerServiceClient secretManagerServiceClient;
+
+	private final GcpProjectIdProvider projectIdProvider;
+
+	public SecretManagerTemplate(
+			SecretManagerServiceClient secretManagerServiceClient,
+			GcpProjectIdProvider projectIdProvider) {
+		this.secretManagerServiceClient = secretManagerServiceClient;
+		this.projectIdProvider = projectIdProvider;
+	}
+
+	/**
+	 * Creates a new secret using the provided {@code secretId} and creates a new version of
+	 * the secret with the provided {@code payload}.
+	 *
+	 * <p>
+	 * If there is already a secret saved in SecretManager with the specified
+	 * {@code secretId}, then it simply creates a new version under the secret with
+	 * the secret {@code payload}.
+	 *
+	 * @param secretId the secret ID of the secret to create.
+	 * @param payload the secret payload; supported payload types: (UTF-8 encoded) String and
+	 *     byte[].
+	 */
+	public void createSecret(String secretId, Object payload) {
+		if (!secretExists(secretId)) {
+			createSecret(secretId);
+		}
+
+		createNewSecretVersion(secretId, payload);
+	}
+
+	/**
+	 * Gets the secret payload of the specified {@code secretId} at version
+	 * {@code versionName}.
+	 *
+	 * @param secretId unique identifier of your secret in Secret Manager.
+	 * @param versionName which version of the secret to load. The version can be a version
+	 *     number as a string (e.g. "5") or an alias (e.g. "latest").
+	 * @return The secret payload as String
+	 */
+	public String getSecretString(String secretId, String versionName) {
+		return getSecretVersion(secretId, versionName).toStringUtf8();
+	}
+
+	/**
+	 * Gets the secret payload of the specified {@code secretId} at version
+	 * {@code versionName}.
+	 *
+	 * @param secretId unique identifier of your secret in Secret Manager.
+	 * @param versionName which version of the secret to load. The version can be a version
+	 *     number as a string (e.g. "5") or an alias (e.g. "latest").
+	 * @return The secret payload as byte[]
+	 */
+	public byte[] getSecretPayload(String secretId, String versionName) {
+		return getSecretVersion(secretId, versionName).toByteArray();
+	}
+
+	private ByteString getSecretVersion(String secretId, String versionName) {
+		SecretVersionName secretVersionName = SecretVersionName.of(
+				this.projectIdProvider.getProjectId(),
+				secretId,
+				versionName);
+
+		AccessSecretVersionResponse response = secretManagerServiceClient.accessSecretVersion(secretVersionName);
+
+		return response.getPayload().getData();
+	}
+
+	/**
+	 * Returns the lower-level {@link SecretManagerServiceClient} client object for making API
+	 * calls to Secret Manager service.
+	 *
+	 * <p>
+	 * Useful for executing more advanced use-cases that are not covered by
+	 * {@link SecretManagerTemplate}.
+	 *
+	 * @return the {@link SecretManagerServiceClient} client object.
+	 */
+	public SecretManagerServiceClient getSecretManagerServiceClient() {
```

Comment:
Why do we need to expose it?
It's a bean in the application context anyway.

Reply:
Done. Removed.